### ISSUE TITLE
DSSM training with cross entropy loss

### ISSUE THREAD

**Issue by ZeonTrevor** [faneshion/MatchZoo#177]
How would I configure MatchZoo to train the DSSM model using cross entropy loss for ranking?

I notice that in the losses/rank_losses.py file there is a definition of rank_crossentropy_loss, but in the inputs/pair_generator.py file I don't see any PairGenerator that would handle/generate the required input needed for this.

**Comment by bwanglzu** [comment]
An easy but dirty way would be change [line 142](https://github.com/faneshion/MatchZoo/blob/e8209d39933a737db693d926df20c498ce30334b/matchzoo/main.py#L142): 
```python
model.compile(optimizer=optimizer, loss=loss)
```
to
```
model.compile(optimizer=optimizer, loss='categorical_crossentropy')
```
Or change the model configuration in `dssm_[DATASETNAME].config`.

**Comment by ZeonTrevor** [comment]
So this would work if the task is for classification, and an example config is also present in the [toy_example](https://github.com/faneshion/MatchZoo/blob/master/examples/toy_example/config/dssm_classify.config). I have edited my initial comment to make it clear that the task is for ranking as discussed in the respective DSSM, CDSSM and DUET papers. 

To set the loss to 'rank_crossentropy_loss', it is clear from the implementation in [rank_losses.py](https://github.com/faneshion/MatchZoo/blob/master/matchzoo/losses/rank_losses.py):

```
"losses": [ 
      {
       "object_name": "rank_crossentropy_loss" ,
       "object_params": {
            "neg_num": 4
       }
    }
  ],
```

But, in this case the input training instance would be of the form:
(query, relevant_doc, non-relevant_doc1,...., non-relevant_doc4)

I don't see any particular PairGenerator in [pair_generator.py](https://github.com/faneshion/MatchZoo/blob/master/matchzoo/inputs/pair_generator.py) that supports this kind of input.

**Comment by bwanglzu** [comment]
@ZeonTrevor Yes, now the question is clear. It's true that the original paper (such as DSSM/CDSSM) sample 4 irrelevant document together with 1 relevant documents for training, how do we handle this? @faneshion @yangliuy .

**Comment by bwanglzu** [comment]
@faneshion can you take a look?

**Comment by faneshion** [comment]
@ZeonTrevor In this version, we have not implemented generator for `cross_entropy_loss`. As we are working on the MatchZoo 2.0, we will add it before long.
